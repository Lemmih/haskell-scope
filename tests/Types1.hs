{-# LANGUAGE ExplicitForAll #-}
module Types1 where

x :: a
x = x

y :: forall b. b
y = y

z :: Show c => c
z = z

f :: a -> b -> a
f = f

g :: a -> forall a. a
g = g
