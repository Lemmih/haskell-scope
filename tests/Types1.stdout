Scope errors:
  Class not in scope:
    z :: [4mShow[0m c => c

Definitions:
  Definition 1, from Types1.decl1:
    [4mx[0m = x
  Definition 2, from Types1.decl2:
    y :: forall [4mb[0m. b
  Definition 3, from Types1.decl3:
    [4my[0m = y
  Definition 4, from Types1.decl5:
    [4mz[0m = z
  Definition 5, from Types1.decl7:
    [4mf[0m = f
  Definition 6, from Types1.decl8:
    g :: a -> forall [4ma[0m. a
  Definition 7, from Types1.decl9:
    [4mg[0m = g

Use sites:
  Definition used: 1
    [4mx[0m :: a
  Definition used: Types1.decl0.tv
    x :: [4ma[0m
  Definition used: 1
    x = [4mx[0m
  Definition used: 3
    [4my[0m :: forall b. b
  Definition used: 2
    y :: forall b. [4mb[0m
  Definition used: 3
    y = [4my[0m
  Definition used: 4
    [4mz[0m :: Show c => c
  Definition used: Types1.decl4.tv
    z :: Show [4mc[0m => c
  Definition used: Types1.decl4.tv
    z :: Show c => [4mc[0m
  Definition used: 4
    z = [4mz[0m
  Definition used: 5
    [4mf[0m :: a -> b -> a
  Definition used: Types1.decl6.tv
    f :: [4ma[0m -> b -> a
  Definition used: Types1.decl6.tv
    f :: a -> [4mb[0m -> a
  Definition used: Types1.decl6.tv
    f :: a -> b -> [4ma[0m
  Definition used: 5
    f = [4mf[0m
  Definition used: 7
    [4mg[0m :: a -> forall a. a
  Definition used: Types1.decl8.tv
    g :: [4ma[0m -> forall a. a
  Definition used: 6
    g :: a -> forall a. [4ma[0m
  Definition used: 7
    g = [4mg[0m

Exports:
  f, Value from Types1
  g, Value from Types1
  x, Value from Types1
  y, Value from Types1
  z, Value from Types1
